noir-projects/aztec-nr/aztec/src/note/note_header.nr

```diff
@@ -31,8 +32,12 @@ impl NoteHeader {
     }
 }
 
-impl Serialize<4> for NoteHeader {
-    fn serialize(self) -> [Field; 4] {
+impl Serialize<NOTE_HEADER_LENGTH> for NoteHeader {
+    /// The following method is used by implementations of the Serialize trait for notes --> the implementation
+    // of the Serialize trait for notes needs to be implemented when the note is passed as an argument to a contract
+    // function --> in that situation the serialize method is called by aztec-nr when computing an arguments hash.
+    fn serialize(self) -> [Field; NOTE_HEADER_LENGTH] {
+        // Note: If you change this function don't forget to update implementations of Serialize trait for notes.
```

Comment:
The Serialize trait is only implemented for a ValueNote because it's passed as an argument to a contract function in 1 contract. So Serialize trait of value note would have to be updated.

Reply:
Expanded the comment in https://github.com/AztecProtocol/aztec-packages/pull/7618/commits/1fe89f250487d672bf2bc2128dce68e69b773613